app/models/manageiq/providers/ibm_cloud/inventory/collector/power_virtual_servers/target_collection.rb

```diff
@@ -0,0 +1,110 @@
+class ManageIQ::Providers::IbmCloud::Inventory::Collector::PowerVirtualServers::TargetCollection < ManageIQ::Providers::IbmCloud::Inventory::Collector::PowerVirtualServers
+  def initialize(_manager, _target)
+    super
+
+    parse_targets!
+  end
+
+  def availability_zones
+    []
+  end
+
+  def images
+    @images ||= begin
+      references(:miq_templates).map do |ems_ref|
+        images_api.pcloud_cloudinstances_images_get(cloud_instance_id, ems_ref)
+      end
+    end
+  end
+
+  def flavors
+    []
+    #TODO: Check if references contains :flavors
+    ## @system_pools ||= system_pools_api.pcloud_systempools_get(cloud_instance_id)
+    ## @sap_profiles ||= sap_api.pcloud_sap_getall(cloud_instance_id).profiles || []
+  end
+
+  def cloud_volume_types
+    []
+    #TODO: Check if references contains :cloud_volume_types
+    ## @storage_types ||= storage_capacity_api.pcloud_storagecapacity_types_getall(cloud_instance_id)
+  end
+
+  def volumes
+    @volumes ||= begin
+      references(:cloud_volumes).map do |ems_ref|
+        volumes_api.pcloud_cloudinstances_volumes_get(cloud_instance_id, ems_ref)
+      end
```

Comment:
@jaywcarman one thing to keep in mind here is if a targeted refresh is requested when something is deleted, then these `GET` calls will return a 404.  Not sure if this SDK will raise an exception (that's how the VPC side handles non `200` results) or return nil.

You probably will have to rescue an API exception, return `nil`, and `.compact` the list.  This will lead to the record being deleted from the database which is exactly what we want rather than an exception causing the refresh to fail.

Reply:
Thanks for the tip @agrare! I should also include a 404 response in the specs for this situation...

Reply:
@agrare I implemented this in the commit I just force pushed.